src/odemis/driver/test/technolution_test.py

```diff
@@ -1517,5 +1486,73 @@ def test_get_and_subscribe(self):
         self.assertEqual(field_images[0] * field_images[1], self.counter)
         self.assertEqual(field_images[0] * field_images[1], self.counter2)
 
+    def test_error_get(self):
+        """Test that exceptions are raised if wrong settings are not caught in the wrapper, but on the ASM."""
+
+        self.MPPC.filename.value = time.strftime("testing_megafield_id-%Y-%m-%d-%H-%M-%S")
+        dataflow = self.MPPC.data
+
+        init_cellCompleteRes = self.MPPC.cellCompleteResolution.value
+        # force a value on a VA which will raise an error on the ASM
+        # use ._value in order to circumvent the setter checks
+        self.MPPC.cellCompleteResolution._value = (1100, 1100)  # max value allowed on HW is 1000; force it!
+
+        # raise exception due to wrong value in settings uploaded to ASM during the start of the acquisition
+        with self.assertRaises(Exception):
+            image = dataflow.get()
+
+        # now check that with a correct value an acquisition can be performed again
+        self.MPPC.cellCompleteResolution.value = init_cellCompleteRes
+        image = dataflow.get()
+        self.assertIsInstance(image, model.DataArray)
+
+    def test_error_subscribe(self):
+        """Test that exceptions are raised if wrong settings are not caught in the wrapper, but on the ASM.
+        Additionally, check that subscribers are properly unsubscribed after an exception occurred."""
+
+        self.MPPC.filename.value = time.strftime("testing_megafield_id-%Y-%m-%d-%H-%M-%S")
+        dataflow = self.MPPC.data
+
+        img_queue_1 = queue.Queue()
+        img_queue_2 = queue.Queue()
+
+        def image_received_1(dataflow, da):
+            img_queue_1.put(da)
+            logging.debug("image 1 received")
```

Comment:
The message is a little bit confusing. I'd suggest "Image received by subscriber 1".